src/include/storage/data_table.h

```diff
@@ -236,6 +239,8 @@ class DataTable {
   // The block compactor elides transactional protection in the gather/compression phase and
   // needs raw access to the underlying table.
   friend class BlockCompactor;
+  friend class Checkpoint;
```

Comment:
Comments on why this was made a friend class would be nice (just like for the BlockCompactor)